src/taskRunner.ts

```diff
@@ -103,7 +103,7 @@ class CustomTaskTerminal implements Pseudoterminal {
       }
     );
 
-    this.callback(data);
+    this.callback(data, "Compile Error");
```

Comment:
Hmm this message only appears if the preview isn't already open.

can we also return the error from d2Tasks.compile and pass it to this callback for the exact message?

I can find the error message in the terminal here: but d2 needs to add line numbers to this error for it to show up in the editor.
![Screen Shot 2023-03-02 at 5 16 10 PM](https://user-images.githubusercontent.com/85081687/222607659-10d5cb42-7a52-471a-8bb7-9bfa65c5aa0d.png)

Reply:
This was intentional.  We leave the last 'good' compile up, but on the first 'show', if there are errors, there is no previous 'good' render to show.

At least a line number is required, and possibly a column, I thought I had a fix for this, but there are some bugs in vscode dealing with multiple problemMatchers, so I couldn't just add one that grabs the filename and error.  At the moment, one size has to fit all.

Reply:
ok that makes sense, but we want to have some indication of a problem if the terminal is collapsed. hmm can we give it something like line 1 col 1 if we don't have a line number, that way the error can still be shown in the editor?

Reply:
I'll put the whole error message in.

Reply:
OK, fixed this up to better show the errors.  We could probably fix this even better with an overlay that pops up on error, but that is a nice to have vs must have.

Found a bug in our problemMatcher too, and that led to another vscode bug (https://github.com/microsoft/vscode/issues/133013#issue-995134699).  I'm thinking problemMatchers are a problem...